### PR TITLE
fix: send notification only once

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Fix workflow controller sent notification multiple times.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/workflow/controller/handlers/workflowrun/handler.go [hunk 1]
```diff
@@ -172,7 +185,15 @@ func (h *Handler) ObjectDeleted(obj interface{}) {
 // * notification endpoint is configured
 // * without notification sent label
 func (h *Handler) sendNotification(wfr *v1alpha1.WorkflowRun) error {
-	if meta.LabelExists(wfr.Labels, meta.LabelWorkflowRunNotificationSent) {
+	h.notifyLock.Lock(wfr.Name)
+	defer h.notifyLock.Unlock(wfr.Name)
```
Comment (cd1989) [inline pkg/workflow/controller/handlers/workflowrun/handler.go]: Seems no need to use lock, controller handles event in a single thread one by one.

Comment (zhujian7) [reply pkg/workflow/controller/handlers/workflowrun/handler.go]: Have fixed, PTAL @cd1989

**Pull request merged**